internal/server/singleprocess/state/project_test.go

```diff
@@ -64,6 +64,51 @@ func TestProject(t *testing.T) {
 		}
 	})
 
+	t.Run("Put does not modify applications", func(t *testing.T) {
+		require := require.New(t)
+
+		const name = "AbCdE"
+		ref := &pb.Ref_Project{Project: name}
+
+		s := TestState(t)
+		defer s.Close()
+
+		// Set
+		proj := serverptypes.TestProject(t, &pb.Project{Name: name})
+		err := s.ProjectPut(proj)
+		require.NoError(err)
+		_, err = s.AppPut(serverptypes.TestApplication(t, &pb.Application{
+			Project: ref,
+		}))
+		require.NoError(err)
+
+		// Get exact
+		{
+			resp, err := s.ProjectGet(&pb.Ref_Project{
+				Project: "AbCdE",
+			})
+			require.NoError(err)
+			require.NotNil(resp)
+			require.False(resp.RemoteEnabled)
+			require.Len(resp.Applications, 1)
```

Comment:
Given this is fixing a bug that would blow away all applications, should the test have more than 1 Applications that we ensure are still around after the Put?

Reply:
Sure, doesn't hurt!